src/models/TransactionModel.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  DataTypes,
+  Model,
+  InferAttributes,
+  InferCreationAttributes,
+  Association,
+  NonAttribute,
+  CreationOptional
+} from 'sequelize'
+import { sequelize } from '../db/connection'
+
+export class Transaction extends Model<
+  InferAttributes<Transaction>,
+  InferCreationAttributes<Transaction>
+> {
+  declare id: CreationOptional<number>
+  declare status: 'pending' | 'reversed' | 'closed'
+  declare type: 'purchase' | 'sale'
```

Comment:
please define these as interfaces(enums) and use them here, since we will be using them elsewhere 

here is an example 
```js
export enum TransactionStatus {
  Pending = 'pending',
  Confirmed = 'confirmed',
  Closed = 'closed',
  Declined = 'declined',
  Rejected = 'rejected',
}
```